src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/NewRule.cs

```diff
@@ -0,0 +1,47 @@
+using DotNetProjectFile.Text;
+
+namespace DotNetProjectFile.Analyzers.MsBuild;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class NewRule() : MsBuildProjectFileAnalyzer(Rule.NewRule)
+{
+    private readonly ImmutableArray<string> Knowns = [.. NodeFactory.KnownNodes.Select(x => x.ToUpperInvariant())];
+
+    protected override void Register(ProjectFileAnalysisContext<MsBuildProject> context)
+    {
+        Walk(context.File, context);
+    }
+
+    private void Walk(Node node, ProjectFileAnalysisContext<MsBuildProject> context)
+    {
+        if (node is Unknown && Suggestion(node) is { Length: > 0 } suggestion)
+        {
+            context.ReportDiagnostic(Descriptor, node, node.LocalName, suggestion);
+        }
+
+        foreach (var child in node.Children)
+        {
+            Walk(child, context);
+        }
+    }
+
+    private string? Suggestion(Node node)
+    {
+        var name = node.LocalName.ToUpperInvariant();
+        string? suggestion = null;
+        var best = name.Length / 2;
```

Comment:
Maybe also good to document this minimum difference before being reported in the documentation somewhere.